tahrir_api/dbapi.py

```diff
@@ -916,6 +916,40 @@ def add_assertion(self, badge_id, person_email, issued_on, issued_for=None):
 
         return False
 
+    @autocommit
+    def remove_assertion(self, badge_id, person_email):
+        """
+        Remove an assertion (revoke a badge) from the database
+
+        :type badge_id: str
+        :param badge_id: ID of the badge to be revoked
+
+        :type person_email: str
+        :param person_email: Email of the Person to revoke the badge from
+
+        :returns: tuple of (person_email, badge_id) if successful, False otherwise
+        """
+
+        if not self.person_exists(email=person_email):
+            return False
+
+        if not self.badge_exists(badge_id):
+            return False
+
+        person = self.get_person(person_email)
+
+        assertion = (
+            self.session.query(Assertion).filter_by(person_id=person.id, badge_id=badge_id).first()
+        )
+
+        if not assertion:
+            return False
```

Comment:
Since you're returning `False` if the assertion is not found, it will make no difference if you don't check that the badge exists first. If the badge does not exist and you don't check for it first, the assertion query will find nothing and you'll return False anyway.